File: Deepskilling/m-1(design pattern)/Logger.java
public class Logger {

    // Stores the single Logger object
    private static Logger instance;

    // Private constructor
    // Prevents object creation using new Logger()
    private Logger() {
        System.out.println("Logger Created");
    }

    // Returns the same Logger object every time
    public static Logger getInstance() {

        if (instance == null) {
            instance = new Logger();
        }

        return instance;
    }

    public void log(String message) {
        System.out.println("LOG: " + message);
    }
}

File: Deepskilling/Module -1 design pattern and principle/Singleton pattern exercise-01/Logger.cs
// Singleton Class

public class Logger
{
    // Stores the single object
    private static Logger instance;

    // Private constructor
    // Prevents object creation using new Logger()
    private Logger()
    {
        Console.WriteLine("Logger Created");
    }

    // Returns the same object every time
    public static Logger GetInstance()
    {
        if (instance == null)
        {
            instance = new Logger();
        }

        return instance;
    }

    public void Log(string message)
    {
        Console.WriteLine("LOG: " + message);
    }
}